Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -37,27 +43,134 @@ YUI.add('ez-roleserversideview', function (Y) {
-                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button);
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
+                    roleName: button.getAttribute('data-role-name'),
+                    afterUpdateCallback: unsetLoading,
+                };
 
             e.preventDefault();
             this._uiSetUDWButtonLoading(button);
+            this._fireContentDiscover(button, unsetLoading, udwConfigData);
+        },
+
+        /**
+         * tap event handler for assigning role with subtree limitation. It launches the
+         * universal discovery widget a first time so that the user can pick a location.
+         *
+         * @method _pickLocationLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickSubtreeLimitation: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                that = this;
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-limit-subtree-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    multiple: true,
+                    contentDiscoveredHandler: function(e) {
+                        that._setSubtreeLimitation(button, this, e);
+                    }
+                },
+            });
+        },
+
+        /**
+         * Launch a second time the UDW so that the user can pick some contents.
+         * The UDW config datas will be filled by the role infos and the locations of the subtree limitation chosen before.
+         *
+         * @method _pickSubtreeLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _setSubtreeLimitation: function (button, udView, e) {
+            var unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                that = this,
+                selectedLocationsIds = Y.Array.map(e.selection, function(struct) {
+
+                return struct.location.get('id');
+            }),
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
+                    roleName: button.getAttribute('data-role-name'),
+                    afterUpdateCallback: unsetLoading,
+                    limitationType: 'Subtree',
+                    subtreeIds: selectedLocationsIds,
+                },
+                udwAfterActiveChangeEvent = udView.onceAfter('activeChange', function() {
+                    udwAfterActiveChangeEvent.detach();
+                        setTimeout(function() {
+                            that._fireContentDiscover(button, unsetLoading, udwConfigData);
```

Comment:
That's an option. You can also set the context of the `onceAfter` function with its third parameter and indeed bind the `setTimeout` functon.